cmd/snaplock/runinhibit/inhibit.go

```diff
@@ -308,55 +308,59 @@ var newTicker = func(interval time.Duration) ticker {
 // NOTE: A snap without a hint file is considered not inhibited and a nil FileLock is returned.
 //
 // NOTE: It is the caller's responsibility to release the returned file lock.
-var WaitWhileInhibited = func(snapName string, notInhibited func() error, inhibited func(hint Hint, inhibitInfo *InhibitInfo) (cont bool, err error), interval time.Duration) (flock *osutil.FileLock, retErr error) {
+var WaitWhileInhibited = func(snapName string, notInhibited func() error, inhibited func(hint Hint, inhibitInfo *InhibitInfo) (cont bool, err error), interval time.Duration) (flock *osutil.FileLock, err error) {
 	ticker := newTicker(interval)
 
+	// Release lock if we return early with an error
+	defer func() {
+		// Keep lock if no errors occur
+		if err != nil && flock != nil {
+			flock.Close()
+			flock = nil
+		}
+	}()
+
 	for {
-		flock, err := osutil.OpenExistingLockForReading(HintFile(snapName))
+		flock, err = osutil.OpenExistingLockForReading(HintFile(snapName))
+		// We must return flock alongside errors so that cleanup defer can close it.
 		if os.IsNotExist(err) {
 			if err := notInhibited(); err != nil {
-				return nil, err
+				return flock, err
```

Comment:
this should stay nil as we haven't opened the flock on this path

Reply:
thanks for catching this, updated!